Some seemingly legit test failures so far on Android:

```
   Restored /Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/RunOnAndroidmauiblazornet80Release.csproj (in 2.24 sec).
 /Users/builder/azdo/_work/2/s/bin/dotnet/sdk/8.0.100-rtm.23506.1/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.RuntimeIdentifierInference.targets(311,5): message NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotnet-support-policy [/Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/RunOnAndroidmauiblazornet80Release.csproj::TargetFramework=net8.0-android]
 /Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/Platforms/Android/Instrumentation.cs(17,30): error CS8765: Nullability of type of parameter 'arguments' doesn't match overridden member (possibly because of nullability attributes). [/Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/RunOnAndroidmauiblazornet80Release.csproj::TargetFramework=net8.0-android]
 /Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/Platforms/Android/Instrumentation.cs(29,39): error CS8600: Converting null literal or possible null value to non-nullable type. [/Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/RunOnAndroidmauiblazornet80Release.csproj::TargetFramework=net8.0-android]
 /Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/Platforms/Android/Instrumentation.cs(32,33): error CS8604: Possible null reference argument for parameter 'packageContext' in 'Intent Intent.SetClassName(Context packageContext, string className)'. [/Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/RunOnAndroidmauiblazornet80Release.csproj::TargetFramework=net8.0-android]
 /Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/Platforms/Android/Instrumentation.cs(34,40): error CS8600: Converting null literal or possible null value to non-nullable type. [/Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/RunOnAndroidmauiblazornet80Release.csproj::TargetFramework=net8.0-android]
 Build FAILED.
 /Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/Platforms/Android/Instrumentation.cs(17,30): error CS8765: Nullability of type of parameter 'arguments' doesn't match overridden member (possibly because of nullability attributes). [/Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/RunOnAndroidmauiblazornet80Release.csproj::TargetFramework=net8.0-android]
 /Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/Platforms/Android/Instrumentation.cs(29,39): error CS8600: Converting null literal or possible null value to non-nullable type. [/Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/RunOnAndroidmauiblazornet80Release.csproj::TargetFramework=net8.0-android]
 /Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/Platforms/Android/Instrumentation.cs(32,33): error CS8604: Possible null reference argument for parameter 'packageContext' in 'Intent Intent.SetClassName(Context packageContext, string className)'. [/Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/RunOnAndroidmauiblazornet80Release.csproj::TargetFramework=net8.0-android]
 /Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/Platforms/Android/Instrumentation.cs(34,40): error CS8600: Converting null literal or possible null value to non-nullable type. [/Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/RunOnAndroidmauiblazornet80Release.csproj::TargetFramework=net8.0-android]
     0 Warning(s)
     4 Error(s)
```

--- a/src/TestUtils/src/Microsoft.Maui.IntegrationTests/TestResources/TemplateLaunchInstrumentation.cs
+++ b/src/TestUtils/src/Microsoft.Maui.IntegrationTests/TestResources/TemplateLaunchInstrumentation.cs
@@ -8,13 +8,13 @@ namespace mauitemplate
     [Instrumentation(Name = "com.microsoft.mauitemplate.Launch")]
     public class TemplateLaunchInstrumentation : Instrumentation
     {
-        protected TemplateLaunchInstrumentation ()
-        {}
+        protected TemplateLaunchInstrumentation()
+        { }
 
-        protected TemplateLaunchInstrumentation (IntPtr handle, JniHandleOwnership transfer) : base (handle, transfer)
-        {}
+        protected TemplateLaunchInstrumentation(IntPtr handle, JniHandleOwnership transfer) : base(handle, transfer)
+        { }
 
-        public override void OnCreate(Bundle arguments)
+        public override void OnCreate(Bundle? arguments)
         {
             base.OnCreate(arguments);
             Start();
@@ -24,14 +24,14 @@ namespace mauitemplate
         {
             base.OnStart();
 
-            Bundle results = new Bundle ();
+            Bundle results = new Bundle();
             var activityName = "com.microsoft.mauitemplate.MainActivity";
-            ActivityMonitor monitor = AddMonitor(activityName, null, false);
+            var monitor = AddMonitor(activityName, null, false);
             Intent intent = new Intent(Intent.ActionMain);
             intent.SetFlags(ActivityFlags.NewTask);
-            intent.SetClassName(TargetContext, activityName);
+            intent.SetClassName(TargetContext!, activityName);
             StartActivitySync(intent);
-            Activity currentActivity = WaitForMonitor(monitor);
+            var currentActivity = WaitForMonitor(monitor);
             var resultCode = currentActivity is not null ? Result.Ok : Result.Canceled;
             results.PutString("return-code", resultCode.ToString("D"));
             Finish(resultCode, results);

--- a/src/TestUtils/src/Microsoft.Maui.IntegrationTests/TestResources/TemplateLaunchInstrumentation.cs
+++ b/src/TestUtils/src/Microsoft.Maui.IntegrationTests/TestResources/TemplateLaunchInstrumentation.cs
@@ -1,4 +1,5 @@
-﻿using Android.App;
+﻿#nullable enable
+using Android.App;
 using Android.Content;
 using Android.OS;
 using Android.Runtime;

